cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultServicesManagerImpl.java

```diff
@@ -140,7 +141,7 @@ public RegisteredService findServiceBy(final long id) {
      *
      * @return the tree set
      */
-    public TreeSet<RegisteredService> convertToTreeSet() {
+    public Collection<RegisteredService> convertToTreeSet() {
```

Comment:
I would personally suggest to use an interface specialization of a `Set`, namely `SortedSet` (as `TreeSet` is an implementation of it) and rename the method to `convertToSortedSet`. This will keep the right level of abstraction and make IDEA happy as an additional benefit.